utils/fetchLSP8Assets.ts

```diff
@@ -1,98 +1,80 @@
-import { LSP4DigitalAssetJSON } from '@lukso/lsp-factory.js/build/main/src/lib/interfaces/lsp4-digital-asset'
 import LSP8IdentifiableDigitalAsset from '@lukso/lsp-smart-contracts/artifacts/LSP8IdentifiableDigitalAsset.json'
 
-import { LSP8Asset } from '@/types/assets'
-import { PROVIDERS } from '@/types/enums'
+import { Asset } from '@/types/assets'
+import { LSP8IdentifiableDigitalAsset as LSP8IdentifiableDigitalAssetInterface } from '@/types/contracts/LSP8IdentifiableDigitalAsset'
 
-export const fetchLSP8Assets = async (
-  assetAddress: Address,
+export const fetchLsp8Assets = async (
+  address: Address,
   profileAddress: Address
 ) => {
   const { contract } = useWeb3(PROVIDERS.RPC)
+  const assets: Asset[] = []
 
-  const lsp8Contract = contract(
+  const lsp8Contract = contract<LSP8IdentifiableDigitalAssetInterface>(
     LSP8IdentifiableDigitalAsset.abi as any,
-    assetAddress
+    address
   )
   const tokenSupply = await lsp8Contract.methods.totalSupply().call()
 
   // profile can have few ids of same LSP8 asset
-  const tokensIds = (await lsp8Contract.methods
-    .tokenIdsOf(profileAddress)
-    .call()) as string[]
+  const tokensIds = await lsp8Contract.methods.tokenIdsOf(profileAddress).call()
 
   if (!tokensIds.length) {
-    return
+    return assets
   }
 
-  const { fetchLSP4Metadata, fetchLSP8Metadata, fetchLSP4Creator } = useErc725()
-  const [collectionName, collectionSymbol, collectionLSP4Metadata] =
-    await fetchLSP4Metadata(assetAddress)
-
-  const newLSP8Assets: LSP8Asset[] = []
+  const { fetchLsp8Metadata, fetchLSP4Creator } = useErc725() // TODO move to utils
+  // nft metadata is the same for all tokens of same asset
+  const [name, symbol, nftMetadata] = await fetchLsp4Metadata(address)
 
   await Promise.all(
     tokensIds.map(async tokenId => {
-      const nftMetadata = await fetchLSP8Metadata(tokenId, assetAddress)
-      const creatorMetadata = await fetchLSP4Creator(assetAddress)
+      const collectionMetadata = (await fetchLsp8Metadata(tokenId, address))
+        .LSP4Metadata
+      const {
+        description,
+        images: metadataImages,
+        icon: metadataIcon,
+        links,
+      } = collectionMetadata
+      const creatorMetadata = await fetchLSP4Creator(address)
+      const {
+        name: creatorName,
+        address: creatorAddress,
+        profileImageUrl: creatorProfileImage,
+      } = creatorMetadata || {}
+      const icon = await getAndConvertImage(metadataIcon, 200)
+      const images: Base64EncodedImage[] = []
 
-      const lsp8AssetObject: LSP8Asset = createLSP8Object(
-        assetAddress,
-        tokenId,
+      for await (const image of metadataImages) {
+        const convertedImage = await getAndConvertImage(image, 400)
+        if (convertedImage) {
+          images.push(convertedImage)
+        }
+      }
+
+      assets.push({
```

Comment:
nit 💅🏼 and just a suggestion. Even though this is async you could just return here the object and since you are using the promiseAll await above you could do:
```const assets = await Promise.all(
     // rest of code
     return assets;
```

Reply:
Thanks for the hint!